tf/examples/Bonsai/bonsai_example.py

```diff
@@ -74,21 +78,28 @@ def main():
 
     helpermethods.dumpCommand(sys.argv, currDir)
 
+    print("Num of classes : ", numClasses)
     # numClasses = 1 for binary case
     bonsaiObj = Bonsai(numClasses, dataDimension,
-                       projectionDimension, depth, sigma)
+                       projectionDimension, depth, sigma, isRegression=isRegression)
 
     bonsaiTrainer = BonsaiTrainer(bonsaiObj,
                                   regW, regT, regV, regZ,
                                   sparW, sparT, sparV, sparZ,
                                   learningRate, X, Y, useMCHLoss, outFile)
 
     sess = tf.InteractiveSession()
-    sess.run(tf.global_variables_initializer())
+    '''
+    'tf.local_variables_initializer()' is required to initialize the local variables used by tf.metrics.mean_absolute_error()
+    '''
+    sess.run(tf.group(tf.global_variables_initializer(), tf.local_variables_initializer()))
```

Comment:
Can't you implement your MAE in utils.py in that case? Isn't it just a broadcast subtraction followed by avg of mods?